stetho/src/main/java/com/facebook/stetho/inspector/protocol/module/CSS.java

```diff
@@ -33,18 +46,276 @@ public void disable(JsonRpcPeer peer, JSONObject params) {
+
+    match.matchingSelectors.add(0);
+
+    CSSRule rule = new CSSRule();
+
+    rule.origin = "regular";
+    rule.selectorList = new SelectorList();
+
+    rule.selectorList.selectors = new ArrayList<>();
+    rule.selectorList.selectors.add(selector);
+
+    rule.style = new CSSStyle();
+    rule.style.cssProperties = new ArrayList<>();
+
+    match.rule = rule;
+
+    rule.style.shorthandEntries = new ArrayList<>();
+
+    mDocument.postAndWait(new Runnable() {
+      @Override
+      public void run() {
+        mDocument.getElementStyles(
+            mDocument.getElementForNodeId(request.nodeId),
+            new StyleAccumulator() {
+              @Override
+              public void store(String name, String value, boolean isDefault) {
+                if (!isDefault) {
+                  CSSProperty property = new CSSProperty();
+                  property.name = name;
+                  property.value = value;
+
+                  match.rule.style.cssProperties.add(property);
+                }
+              }
+            });
+      }
+    });
+
+    result.matchedCSSRules.add(match);
+
+    result.inherited = new ArrayList<>();
+    result.pseudoElements = new ArrayList<>();
+
+    return result;
   }
 
-  private static class GetSupportedCSSPropertiesResponse implements JsonRpcResult {
+  private final class PeerManagerListener extends PeersRegisteredListener {
+    @Override
+    protected synchronized void onFirstPeerRegistered() {
+      mDocument.addRef();
+    }
+
+    @Override
+    protected synchronized void onLastPeerUnregistered() {
+      mDocument.release();
+    }
+  }
+
+  private static class CSSComputedStyleProperty {
+    @JsonProperty(required = true)
+    public String name;
+
     @JsonProperty(required = true)
-    public List<CSSPropertyInfo> cssProperties;
+    public String value;
   }
 
-  private static class CSSPropertyInfo {
-    // Incomplete
+  private static class RuleMatch {
+    @JsonProperty
+    public CSSRule rule;
+
+    @JsonProperty
+    public List<Integer> matchingSelectors;
+  }
+
+  private static class SelectorList {
+    @JsonProperty
+    public List<Selector> selectors;
+
+    @JsonProperty
+    public String text;
+  }
+
+  private static class SourceRange {
+    @JsonProperty(required = true)
+    public int startLine;
+
+    @JsonProperty(required = true)
+    public int startColumn;
+
+    @JsonProperty(required = true)
+    public int endLine;
+
+    @JsonProperty(required = true)
+    public int endColumn;
+  }
+
+  private static class Selector {
+    @JsonProperty(required = true)
+    public String value;
+
+    @JsonProperty
+    public SourceRange range;
+  }
+
+  private static class CSSRule {
+    @JsonProperty
+    public String styleSheetId;
+
+    @JsonProperty(required = true)
+    public SelectorList selectorList;
+
+    // this is an enum, and can be one
+    // of the following values:
+    // ["injected", "user-agent", "inspector", "regular"]
```

Comment:
enums are supported directly ... check out `NodeType` which is used by `DOM.Node`. there's a pattern where you implement a `getProtocolValue` method on your enum to supply the JSON serialized form. Hopefully it works with strings.